GitCommands/Git/GitModule.cs

```diff
@@ -2170,20 +2170,31 @@ public IReadOnlyList<PatchFile> GetInteractiveRebasePatchFiles()
             foreach (string[] parts in todoCommitsInfos)
             {
                 string commitHash = parts[1];
+                string? error = null;
                 CommitData? data = rebasedCommitsRevisions.TryGetValue(commitHash, out GitRevision commitRevision)
                     ? _commitDataManager.CreateFromRevision(commitRevision, null)
-                    : _commitDataManager.GetCommitData(commitHash, out var _);
+                    : _commitDataManager.GetCommitData(commitHash, out error);
 
                 bool isApplying = currentCommitShortHash is not null && commitHash.StartsWith(currentCommitShortHash);
                 isCurrentFound |= isApplying;
 
+                ObjectId objectId;
+                if (data is not null)
+                {
+                    objectId = data.ObjectId;
+                }
+                else
+                {
+                    ObjectId.TryParse(parts[1], out objectId);
+                }
+
                 patchFiles.Add(new PatchFile
                 {
-                    Author = data?.Author,
-                    ObjectId = data?.ObjectId,
-                    Subject = data?.Body,
                     Action = parts[0],
-                    Date = data?.CommitDate.LocalDateTime.ToString(),
+                    ObjectId = objectId,
+                    Subject = data?.Body ?? string.Join(' ', parts.Skip(2)),
```

Comment:
Is Subject or Body needed?

Reply:
Before this PR, the grid was using the body (1st line i.e. subject displayed in the grid and body displayed in the tooltip).
So my goal is more to not break this behavior.
But we could decide that subject is enough and the user will double click to see the whole message (even if personally I found the whole body in the tooltip useful in some rare cases when you rebase some commits with very similar subjects)

Reply:
Not sure to get what you mean but I have added a comment.

Reply:
I meant on line 2195 - or someone may change the code to only get the subject